src/test/java/org/swisspush/redisques/QueueStatsServiceTest.java

```diff
@@ -0,0 +1,224 @@
+package org.swisspush.redisques;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.EventBus;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.swisspush.redisques.util.DequeueStatistic;
+import org.swisspush.redisques.util.DequeueStatisticCollector;
+import org.swisspush.redisques.util.QueueStatisticsCollector;
+import org.swisspush.redisques.util.RedisquesConfiguration;
+import org.swisspush.redisques.exception.RedisQuesExceptionFactory;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.concurrent.Semaphore;
+import java.util.function.BiConsumer;
+
+@RunWith(VertxUnitRunner.class)
+public class QueueStatsServiceTest {
+
+    private Vertx vertx;
+    private EventBus eventBus;
+    private QueueStatsService queueStatsService;
+    private RedisquesConfiguration config;
+    private DequeueStatisticCollector dequeueStatisticCollector;
+
+    /**
+     * Test setup for cases where dequeue stats are enabled.
+     */
+    @Before
```

Comment:
Do not use multiple `@Before` methods. These will all be called for every test and then again when you call them from your test.

Use a single `@Before` method for stuff that is true for all tests and put the rest directly in the test.